src/nuc/selector.py

```diff
@@ -64,16 +96,34 @@ def apply(self, value: Dict[str, Any]) -> Any:
         value
             The dict that this selector should be applied to.
         """
+        match self.target:
+            case SelectorTarget.TOKEN:
+                return self._apply_on_value(self.path, value)
+            case SelectorTarget.CONTEXT:
+                if not self.path:
+                    return None
+                first = self.path[0]
+                context_value = context.get(first)
+                if not context_value:
+                    return None
+                return self._apply_on_value(self.path[1:], context_value)
 
+    @staticmethod
+    def _apply_on_value(path: List[str], value: Dict[str, Any]) -> Any:
         output = value
-        for label in self.path:
+        for label in path:
             output = output.get(label)
             if not output:
```

Comment:
Same here, values like `0`, `False` and `""` would eval to `False` which can be real output value?

Reply:
Nice catch, gotta love implicit conversions

Reply:
because of that I am having love hate relationship with Python 🤓